proto/process/agent.proto

```diff
@@ -938,85 +947,114 @@ enum ResourceRequirementsType {
-    string concurrencyPolicy = 3;
-    bool suspend = 4;
-    int32 successfulJobsHistoryLimit = 5;
-    int32 failedJobsHistoryLimit = 6;
+	string schedule = 1;
+	int64 startingDeadlineSeconds = 2;
+	string concurrencyPolicy = 3;
+	bool suspend = 4;
+	int32 successfulJobsHistoryLimit = 5;
+	int32 failedJobsHistoryLimit = 6;
 }
 
 // reference https://github.com/kubernetes/kubernetes/blob/cb19b56831d54d1d31249949318ef0b07bf00df9/staging/src/k8s.io/api/batch/v1beta1/types.go#L150
 message CronJobStatus {
-    repeated ObjectReference active = 1;
-    int64 lastScheduleTime = 2;
+	repeated ObjectReference active = 1;
+	int64 lastScheduleTime = 2;
 }
 
 // reference https://github.com/kubernetes/kubernetes/blob/cb19b56831d54d1d31249949318ef0b07bf00df9/staging/src/k8s.io/api/batch/v1beta1/types.go#L58
 message CronJob {
-    Metadata metadata = 1;
-    CronJobSpec spec = 2;
-    CronJobStatus status = 3;
-    bytes yaml = 4;
-    repeated string tags = 5;
+	Metadata metadata = 1;
+	CronJobSpec spec = 2;
+	CronJobStatus status = 3;
+	bytes yaml = 4;
+	repeated string tags = 5;
 }
 
 // reference https://github.com/kubernetes/kubernetes/blob/cb19b56831d54d1d31249949318ef0b07bf00df9/staging/src/k8s.io/api/apps/v1/generated.proto#L119
 message DaemonSetSpec {
-    repeated LabelSelectorRequirement selectors = 1;
-    string deploymentStrategy = 2;
-    string maxUnavailable = 3; // int or str
-    int32 minReadySeconds = 4;
-    int32 revisionHistoryLimit = 5;
+	repeated LabelSelectorRequirement selectors = 1;
+	string deploymentStrategy = 2;
+	string maxUnavailable = 3; // int or str
+	int32 minReadySeconds = 4;
+	int32 revisionHistoryLimit = 5;
 }
 
 // reference https://github.com/kubernetes/kubernetes/blob/cb19b56831d54d1d31249949318ef0b07bf00df9/staging/src/k8s.io/api/apps/v1/generated.proto#L152
 message DaemonSetStatus {
-    int32 currentNumberScheduled = 1;
-    int32 numberMisscheduled = 2;
-    int32 desiredNumberScheduled = 3;
-    int32 numberReady = 4;
-    int32 updatedNumberScheduled = 6;
-    int32 numberAvailable = 7;
-    int32 numberUnavailable = 8;
+	int32 currentNumberScheduled = 1;
+	int32 numberMisscheduled = 2;
+	int32 desiredNumberScheduled = 3;
+	int32 numberReady = 4;
+	int32 updatedNumberScheduled = 6;
+	int32 numberAvailable = 7;
+	int32 numberUnavailable = 8;
 }
 
 // reference https://github.com/kubernetes/kubernetes/blob/cb19b56831d54d1d31249949318ef0b07bf00df9/staging/src/k8s.io/api/apps/v1/generated.proto#L66
 message DaemonSet {
-    Metadata metadata = 1;
-    DaemonSetSpec spec = 2;
-    DaemonSetStatus status = 3;
-    bytes yaml = 4;
-    repeated string tags = 5;
+	Metadata metadata = 1;
+	DaemonSetSpec spec = 2;
+	DaemonSetStatus status = 3;
+	bytes yaml = 4;
+	repeated string tags = 5;
+}
+
+// reference https://github.com/kubernetes/kubernetes/blob/cb19b56831d54d1d31249949318ef0b07bf00df9/staging/src/k8s.io/api/apps/v1/generated.proto#L586
+message StatefulSetSpec {
+	int32 replicas = 1;
```

Comment:
I know this is called `replicas` in the kube spec, but it actually means `desiredReplicas`, right?

Reply:
yes I can also rename it directly here to avoid the confusion with the actual `replicas` in the status